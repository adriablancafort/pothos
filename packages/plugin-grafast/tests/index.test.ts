import { printSchema } from 'graphql';
import { schema } from './examples/basic/schema';

describe('grafast', () => {
  it('generates expected schema', () => {
    expect(printSchema(schema)).toMatchSnapshot();
  });
});
