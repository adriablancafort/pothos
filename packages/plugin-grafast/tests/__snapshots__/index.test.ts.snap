// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`grafast > generates expected schema 1`] = `
"type Alien {
  id: ID
}

interface Animal {
  id: ID
}

type Cat implements Animal {
  id: ID
}

type Dog implements Animal {
  id: ID
}

union Entity = Alien | Cat | Dog

type Query {
  addTwoNumbers(a: Int!, b: Int!): Int
  animal(id: String!): Animal
  currentUser: User
  entity(id: String!): Entity
}

"""A user"""
type User {
  bio: String
  friends: [User!]
  id: ID
  name: String
}"
`;
